zerver/tests/test_message_edit.py

```diff
@@ -1596,3 +1598,100 @@ def test_user_group_mention_restrictions_while_editing(self) -> None:
             },
         )
         self.assert_json_success(result)
+
+    def test_move_message_to_new_topic_with_automatic_follow_policy(self) -> None:
+        self.login("iago")
+        hamlet = self.example_user("hamlet")
+        shiva = self.example_user("shiva")
+        stream = self.make_stream("new_stream")
+        self.subscribe(hamlet, stream.name)
+        self.subscribe(shiva, stream.name)
+
+        self.send_stream_message(
+            sender=hamlet,
+            stream_name="new_stream",
+            topic_name="test",
+            content="Never here",
+        )
+        msg_id = self.send_stream_message(
+            sender=hamlet,
+            stream_name="new_stream",
+            topic_name="test",
+            content="Always here",
+        )
+
+        do_change_user_setting(
+            hamlet,
+            "automatically_follow_topics_policy",
+            UserProfile.AUTOMATICALLY_CHANGE_VISIBILITY_POLICY_ON_INITIATION,
+            acting_user=None,
+        )
+        result = self.client_patch(
```

Comment:
We can use `do_edit_message_assert_success`

Reply:
Hm, that's right. I'll let @timabbott comment on this if it is worth the effort.

Reply:
The explicit API call seems totally fine to me.